README.md

```diff
@@ -1,28 +1,96 @@
-# Introduction
+# Nimble Infrastructure Template
 
-Infrastructure (Terraform) configuration template for projects at Nimble.
+An infrastructure template for web applications powered by Terraform.
 
-## Usage
+**Supported cloud:** AWS. We are looking for contributions to implement support for GCP, Heroku, and Azure!
 
-1. Create a folder name and cd into that folder
+**Supported flavors:**
+- A `Basic` flavor is in the plan, but not available yet
+- The `Complete` flavor generates the following infrastructure:
 
-```bash
-mkdir awesome-infra && cd awesome-infra
-```
+![Diagram of the Complete Infrastructure](https://github.com/nimblehq/infrastructure-templates/blob/feature/55-add-documentation/img/diagram_complete.png?raw=true)
+
+# Usage
 
-2. Start the CLI to generate the infrastructure configuration in Terraform
+## Generate a new project
 
 ```bash
 npm install -g @nimblehq/infra-template
 
 nimble-infra generate {project-name}
 ```
 
-* With npx:
+or
+
 ```bash
 npx @nimblehq/infra-template generate {project-name}
 ```
 
+## Deploy your infrastructure
+
+You can deploy your infrastructure with the [terraform CLI](https://learn.hashicorp.com/collections/terraform/cli) (`brew install terraform`), or directly using the [Terraform Cloud](https://cloud.hashicorp.com/products/terraform).
+
+This template remains generic to provide more flexibility.
+A workspace and folder strategy needs to be defined first:
+- What resources need to be global?
+- What resources need to be specific to an environment?
+- What workspaces will you need?
+
+While advanced users won't need it, the next part provides a "starter kit" architecture that suits most of the new projects.
+
+## Starter kit
+
+The starter kit has 3 workspaces:
+- **Shared**, that handles a single ECR (Elastic Container Registry) for the whole infrastructure
+- **Staging** and **Prod**: Both workspaces are similar but use different resource names (using the suffixes `-staging` and `-prod`)
+
+### Step 1, Source Code
+
+In the source code:
+- create a `base` and a `shared` folder.
```

Comment:
I think for these steps, we could let the template does it, like generating the `shared` folder beforehand, so we could avoid some manual step like this, what do you think @malparty ? If that makes sense, we could create an issue for task :D

Reply:
Issue created! :) 
- #85